Sensors/SMCDellSensors/SMIMonitor.hpp

```diff
@@ -195,33 +240,65 @@ class SMIMonitor : public OSObject
 	 *  @return true on sucess
 	 */
 	bool findTempSensors();
-
+	
 	/**
-	 *  Initial sensor update on startup
+	 *  static Thread Entry method
+	 *
+	 *  @param param0 unused
+	 *  @param param1 unused
 	 */
-	bool initialUpdateSensors {false};
-
+	static void staticUpdateThreadEntry(thread_call_param_t param0, thread_call_param_t param1);
+	
 	/**
-	 *  Update sensors values, must be guarded by mainLock
+	 *  Update sensors values
 	 */
-	void updateSensors();
-
+	void updateSensorsLoop();
+	
+	/*
+	 * Handle SMC updates in idle
+	 */
+	void handleSmcUpdatesInIdle(int idle_loop_count);
+	
+	/**
+	 *  SMC update handlers
+	 */
+	void hanldeManualControlUpdate(size_t index, UInt8 *data);
+	void hanldeManualTargetSpeedUpdate(size_t index, UInt8 *data);
+	void handleManualForceFanControlUpdate(UInt8 *data);
+	
 private:
 	int  i8k_smm(SMMRegisters *regs);
 	bool i8k_get_dell_sig_aux(int fn);
-	bool i8k_get_dell_signature(void);
+	bool i8k_get_dell_signature();
 	int  i8k_get_temp(int sensor);
 	int  i8k_get_temp_type(int sensor);
-	int  i8k_get_power_status(void);
+	int  i8k_get_power_status();
 	int  i8k_get_fan_speed(int fan);
 	int  i8k_get_fan_status(int fan);
 	int  i8k_get_fan_type(int fan);
 	int  i8k_get_fan_nominal_speed(int fan, int speed);
 
-public:
 	int  i8k_set_fan(int fan, int speed);
 	int  i8k_set_fan_control_manual(int fan);
 	int  i8k_set_fan_control_auto(int fan);
 };
 
+// Helper definitions
+static constexpr SMC_KEY KeyF0Md = SMC_MAKE_IDENTIFIER('F',0,'M','d');
+static constexpr SMC_KEY KeyF0Tg = SMC_MAKE_IDENTIFIER('F',0,'T','g');
+static constexpr SMC_KEY KeyFS__ = SMC_MAKE_IDENTIFIER('F','S','!',' ');
+
+inline UInt16 swap_value(UInt16 value)
+{
+	return ((value & 0xff00) >> 8) | ((value & 0xff) << 8);
```

Comment:
Why do we have that code actually? I thought VirtualSMC SDK already has the conversion functions...

Reply:
Helper method has been removed and replaced with OSSwapInt16.